### ISSUE TITLE
How to configure for mocha parallel ?

### ISSUE THREAD

**Issue by Shelvak** [sc-forks/solidity-coverage#662]
Hi there, I'm trying to setup my env with parallel testing (because all my tests take about 4 minutes) and I was able to improve the time to 40seconds with last mocha version, but I can't get solidity-coverage run in parallel.

What I'm using:
- Hardhat: 2.6.0
- Mocha: 9.1.1
- solidity-coverage: 0.7.16

Commands:
Parallel: `npx mocha`
Sequential: `npx hardhat test` (or `npx hardhat coverage` to get the coverage)

My `.mocharc.json`: 
```json
{
  "parallel":   true,
  "jobs":       4,
  "slow":       "1500", // 1.5s
  "full-trace": true,
  "bail":       false,
  "inspect":    true,
  "timeout":    20000,
  "exit":       true,
  "require":    ["hardhat/register", "solidity-coverage"] // tried with `solidity-coverage/api` and others but nothing happened
}
```

Already tried with `mocha:` option in the `.solcover.js` but it runs sequentially with `npx hardhat coverage`.

Mocha requiring hardhat/register starts X processes and then run everything:
![image](https://user-images.githubusercontent.com/873323/131226188-15d2a4a5-a3f1-476f-94fa-c269b96471cd.png)

It's more a question than an issue but it would be awesome if we can get coverage working with mocha parallel =)

Cheers.

**Comment by cgewecke** [comment]
Hi @Shelvak. There's isn't any built-in support for parallelism via mocha. There *are* recipes for parallelizing jobs in CI here:
+ [SetProtocol][1]
+ [Synthetix][2]

(These suites are huge - in the 30-40 min range without parallelization)

The technique used there:  
+ partitioning the test suites into several groups by name
+ spinning up a process for each group and running the coverage task
+ aggregating the `coverage.json` files produced for each job using a tool called [istanbul-combine-updated][3] in a final step


[3]: https://www.npmjs.com/package/istanbul-combine-updated?activeTab=readme
[1]: https://github.com/SetProtocol/set-protocol-v2/blob/70f926ffef6d3a428813bd7fd63b9c6d8b1d3974/.circleci/config.yml#L67-L127
[2]: https://github.com/Synthetixio/synthetix/tree/develop/.circleci/src/jobs

**Comment by cgewecke** [comment]
Another note - it's possible to configure the istanbul coverage report folder name in the `.solcover.js` options. This could be useful for a cli solution that's aggregating reports at the end.

**Comment by Shelvak** [comment]
Awesome  cgewecke, thank you so much for the quick response. I think that I could hook the coverage part in each setup fork and try the merge report after everything. If I succeed I'll post here how I did it=)

**Comment by cgewecke** [comment]
Ok great!

**Comment by cgewecke** [comment]
Closing in favor of #691
